src/java/src/main/java/org/wololo/flatgeobuf/PackedRTree.java

```diff
@@ -382,9 +384,8 @@ public static SearchResult search(InputStream stream, int start, int numItems, i
                 if (isLeafNode)
                     searchResult.hits.add(new SearchHit(indexOffset, pos - 1));
                 else
-                    stack.add(new StackItem(indexOffset, level - 1));
+                    queue.add(new StackItem(indexOffset, level - 1));
             }
-            stack.sort((StackItem a, StackItem b) -> (int) (b.nodeIndex - a.nodeIndex));
```

Comment:
This is the killer line. That said, Stack is one of the JDK 1.0 data structures. It's vastly outdated. If you just need something to beheave like a stack ,with no ordering, pick a Deque implementation instead

Reply:
I now think this sort was a mistake, the items are already guaranteed to be in order.